### PR TITLE
remove pairing_id and simplify pairing process

### PR DESCRIPTION
**Pull request by BertKleewein**

### CODE REVIEW

File: docs/azure-monitor-logging.md [hunk 1]
```diff
@@ -11,64 +11,50 @@ A few built-in Azure Monitor fields are overloaded:
...
+Some service app logs are device specific and contain `deviceId` and `runId` values.
+Other service app logs are not device specific and do not contain `deviceId` and `runId` values.
 
 ## Automaticly populated fields for Python traces.
 
+```json
+  {
+    "lineNumber": "291",
+    "fileName": "service.py",
+    "level": "INFO",
```
Comment (drwill-ms) [inline docs/azure-monitor-logging.md:53]: Will this always match the level of the trace in AppInsights?

File: docs/names-and-ids.md [hunk 2]
```diff
@@ -8,8 +8,8 @@ There are a number of names and IDs used in THIEF.  This table describes some of
 | - | - | - |
 | `servicePool` | string | Name of pool that a service app belongs to. |
 | `requestedServicePool` | string | Name of pool that a device app would like to pair with. |
-| `runId` | guid | Guid representing a running executable.|
-| `pairingId` | guid | Guid representing the pairing between a device app and a service app. |
+| `runId` | guid | Guid representing a single run of a device app.  Regenerated every time the device app launches. |
+| `serviceInstance` | guid | Guid representing a single run of a service app.  Regenerated every time the service app launches. |
```
Comment (drwill-ms) [inline docs/names-and-ids.md:12]: Suggest `serviceInstanceId`

File: python/common/azure_monitor.py [hunk 3]
```diff
@@ -11,15 +11,30 @@
...
 _transport = None
 
 
+class CustomDimensionNames(object):
+    OS_TYPE = "osType"
+    SDK_LANGUAGE = "sdkLanguage"
+    SDK_LANGUAGE_VERSION = "sdkLanguageVersion"
+    SDK_VERSION = "sdkVersion"
+
+    SERVICE_INSTANCE = "serviceIntsance"
```
Comment (drwill-ms) [inline python/common/azure_monitor.py:29]: misspelled instance

File: python/device/device.py [hunk 4]
```diff
@@ -472,131 +460,118 @@ def pairing_thread(self, worker_thread_info):
...
                     # already chose someone else.  Ignore this
                     logger.info("Already paired.  Ignoring.")
 
-                elif not pairing_id or not service_run_id:
+                elif not received_run_id or not received_service_instance:
                     # Or maybe something is wrong with the desired properties.  Probably a
                     # service app that goes by different rules. Ignoring it is better than
                     # crashing.
-                    logger.info("pairingId and/or serviceRunId missing.  Ignoring.")
+                    logger.info("deviceRunId and/or serviceRunId missing.  Ignoring.")
```
Comment (drwill-ms) [inline python/device/device.py:521]: `runId`?

**Review by drwill-ms — APPROVED** [review]

File: docs/azure-monitor-logging.md (hunk 1 above)
Comment (BertKleewein) [reply docs/azure-monitor-logging.md:53]: Good question.  This is done by the OpenCensus libraries that we use to link the python logging infrastructure to AI.  I suspect there is a "best effort" mapping but it's probably not perfect, but I haven't dug into this yet.

File: docs/names-and-ids.md (hunk 2 above)
Comment (BertKleewein) [reply docs/names-and-ids.md:12]: Good idea.  Wish I had thought of this sooner.  I'm going to do one more round of renaming and fit this in.

**Pull request merged**